confight.py

```diff
@@ -164,105 +169,129 @@ def find(path):
         return []
     if os.path.isfile(path):
         return [path]
-    return sorted(glob.glob(os.path.join(path, '*')))
+    return sorted(glob.glob(os.path.join(path, "*")))
 
 
 def check_access(path):
     """Return whether a config file or directory can be read"""
     if not path:
         return False
     elif not os.path.exists(path):
-        logger.debug('Could not find %r', path)
+        logger.debug("Could not find %r", path)
         return False
     elif not os.access(path, os.R_OK):
-        logger.error('Could not read %r', path)
+        logger.error("Could not read %r", path)
         return False
     elif os.path.isdir(path) and not os.access(path, os.X_OK):
-        logger.error('Could not list directory %r', path)
+        logger.error("Could not list directory %r", path)
         return False
     elif os.path.isfile(path) and os.access(path, os.X_OK):
-        logger.warning('Config file %r has exec permissions', path)
+        logger.warning("Config file %r has exec permissions", path)
     return True
 
 
-def load_ini(stream):
-    if 'ExtendedInterpolation' in globals():
+def open_text(path):
+    return io.open(path, mode="r", encoding="utf8")
+
+
+def open_bin(path):
+    return io.open(path, mode="rb", encoding="utf8")
+
+
+def load_json(path):
+    with open_text(path) as stream:
+        return json.load(stream, object_pairs_hook=OrderedDict)
+
+
+def load_toml(path):
+    with open_text(path) as stream:
+        return toml.load(stream, _dict=OrderedDict)
+
+
+def load_ini(path):
+    if "ExtendedInterpolation" in globals():
         parser = ConfigParser(interpolation=ExtendedInterpolation())
     else:
         parser = ConfigParser()
-    parser.read_file(stream)
-    return {
-        section: OrderedDict(parser.items(section))
-        for section in parser.sections()
-    }
+    with open_text(path) as stream:
+        parser.read_file(stream)
+    return {section: OrderedDict(parser.items(section)) for section in parser.sections()}
 
 
-FORMATS = ('toml', 'ini', 'json')
+FORMAT_LOADERS = {"json": load_json, "toml": load_toml, "ini": load_ini}
 FORMAT_EXTENSIONS = {
-    'js': 'json',
-    'json': 'json',
-    'toml': 'toml',
-    'ini': 'ini',
-    'cfg': 'ini',
-}
-FORMAT_LOADERS = {
-    'json': lambda *args: json.load(*args, object_pairs_hook=OrderedDict),
-    'toml': lambda *args: toml.load(*args, _dict=OrderedDict),
-    'ini': load_ini
+    "js": "json",
```

Comment:
**question:** Is that really a ubiquitous abbreviation for the `.json` extension?

I think `.js` means JavaScript. Plus a JavaScript file can contain a JSON object, but they’re not necessarily cross-compatible.

I suggest removing this alias.